### ISSUE TITLE
Separate JSON data for each capture. 

### ISSUE THREAD

**Issue by rajshekarreddy97** [Unity-Technologies/com.unity.perception#90]
How do we create separate JSON files for each image?

**Comment by JonathanHUnity** [comment]
Hi @rajshekarreddy97, right now there is no control over the number of images or captures in each JSON file. How will it help you to have different json files for each image?

**Comment by abhishek1222017** [comment]
Hi @JonathanHUnity If we want to create a dataset for training a CNN based object detection model, we need label for each instance of training classes exist in the image. The model search for(image_name.jpeg,image_name.json). In that case, we need individual json file for images. Thank you!

**Comment by JonathanHUnity** [comment]
It sounds like your model has a specific format of json that it needs to take as input. I would suggest looking at our [Dataset Insights](https://github.com/Unity-Technologies/datasetinsights) python library, which gives examples of how to load the dataset generated from Unity into python and translate it into a form needed to train the Faster R-CNN model. 

In the end, we expect that some amount of data transformation will be required to take the output from Unity and train a model, depending on the specific model you are working with.

**Comment by abhishek1222017** [comment]
Thank you @JonathanHUnity for reply. I would like to clarify our problem better way. Here our problem is not to get json file in a specific format, but individual json file for each image. Currently @rajshekarreddy97 is able to generate one single json file for a sequence of images where json containing bounding box location of each object of interest. I was wondering Instead if we are able to get same data but in different different json files. Thank you in advanced.

**Comment by JonathanHUnity** [comment]
This is not something we have at the moment. If you would like to fork our package, you can find the control for the method we use to split files here: https://github.com/Unity-Technologies/com.unity.perception/blob/master/com.unity.perception/Runtime/GroundTruth/SimulationState.cs#L71

I've also added public control over this value to our list. Hope that helps!

**Comment by adason** [comment]
@abhishek1222017
There was a proposal in #19 to change dataset schema so that each annotation are stored as separate json file for each capture. Do you think this is close to what you expect?

**Comment by rajshekarreddy97** [comment]
@adason Yes, this is exactly what we were looking for.

**Comment by Gowtham171996** [comment]
Hi @adason 
I had a similar requirement and I did it.  Can you advise how can I push for review to you guys?

**Comment by HasnainKhanNiazi** [comment]
> @abhishek1222017
> There was a proposal in #19 to change dataset schema so that each annotation are stored as separate json file for each capture. Do you think this is close to what you expect?

Hello adason, I am working on Unity Perception to generate dataset for semantic segmentation and I want single json annotation file along with maks for each image. I saw the commit that you mentioned but I am not sure how can I use it? A little guidance will be quite helpful. Thanks

**Comment by VincentGuigui** [comment]
If you derive a class from FixedLengthScenario.cs, you can hook you to the onDestroy() method and add a custom transformation for the captures.json file.
```
JObject.Parse(File.ReadAllText(captureFilePath));
foreach (JObject jsonCaptureData in jsonCaptureDataArray)
{
    JArray jsonAnnotationDataArray = jsonCaptureData.Value<JArray>("annotations");
...
```

**Comment by StevenBorkman** [comment]
Hey all! The Perception team just released 🎉 Perception 1.0 🎉, a major update to the toolset! In this release we have added the SOLO format, our new dataset that departs in several ways from the previous output format. This includes a single capture file and artifacts for each frame.

There are a bunch of cool features to check out such as ray tracing, depth, normals, and more! Here is the full list of changes: [Perception 1.0 Changelog](https://github.com/Unity-Technologies/com.unity.perception/blob/main/com.unity.perception/CHANGELOG.md)
